### PR TITLE
Draft: SC4-1-1-unique-id

### PR DESCRIPTION
**Pull request by annethyme**
Rewrote unique-id rule to match new ACT Format

### CODE REVIEW

File: _drafts/SC4-1-1-unique-id.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+<div id="my-div1"> This is my first element</div>
+<div id="my-div2"> This is my second element</div>
+<svg id="my-div3"> This is my third element</svg>
+```
+
+```html
+<div id="my-elm"></div>
+<script>
+  var myElm = document.getElementById('my-elm');
+  var shadow = myElm.attachShadow({ move: 'open' });
```
Comment (kasperisager) [inline _drafts/SC4-1-1-unique-id.md]: `move` -> `mode`

File: _drafts/SC4-1-1-unique-id.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  shadow.innerHTML = '<b id="my-elm"><slot /></b>';
```
Comment (kasperisager) [inline _drafts/SC4-1-1-unique-id.md]: `'<b id="my-elm"><slot /></b>'` => `'<b id="my-elm"><slot></slot></b>'`

File: _drafts/SC4-1-1-unique-id.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+## Test procedure
+
+### Applicability
+
+Any element with an `id` attribute.
+
+### Expectation
+
+All `id` attribute values are unique.
```
Comment (kasperisager) [inline _drafts/SC4-1-1-unique-id.md]: I'd rephrase it along the lines of:

> Each test target has an `id` attribute that is unique within the [document context](#insert-link-here) of the element.

Comment (kasperisager) [reply _drafts/SC4-1-1-unique-id.md]: The term "document context" might not be the best fit; what I want to express is the union of browsing contexts and shadow hosts.

File: _drafts/SC4-1-1-unique-id.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+```html
+<div id="my-div"> This is my first element</div>
+<div id="my-div"> This is my second element</svg>
+```
+
+```html
+<div id="my-div"> This is my first element</div>
+<svg id="my-div"> This is my second element</svg>
+```
```
Comment (WilcoFiers) [inline _drafts/SC4-1-1-unique-id.md]: This should also test that hidden elements with the same ID is a violation.

**Comment by cpandhi** [comment]
Ids should be unique even if the element is hidden as they could be referenced by aria attributes. The rules test case should address that case.

File: _drafts/SC4-1-1-unique-id.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+There are currently no assumptions.
+
+## Accessibility Support
+
+There are no major accessibility support issues known for this rule.
+
+## Background
+- https://www.w3.org/WAI/WCAG20/quickref/?showtechniques=411#qr-ensure-compat-parses
+- https://www.w3.org/TR/WCAG20-TECHS/H74.html
```
Comment (carlosapaduarte) [inline _drafts/SC4-1-1-unique-id.md]: Not really sure that H74 is needed as background for this one

File: _drafts/SC4-1-1-unique-id.md (hunk 3 above)
Comment (Brynanders) [reply _drafts/SC4-1-1-unique-id.md]: I have changed this now inline with your first comment.

File: _drafts/SC4-1-1-unique-id.md (hunk 1 above)
Comment (Brynanders) [reply _drafts/SC4-1-1-unique-id.md]: Done

File: _drafts/SC4-1-1-unique-id.md [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+<div id="my-div2"> This is my second element</div>
+<svg id="my-div3"> This is my third element</svg>
+```
+
+```html
+<div id="my-elm"></div>
+<script>
+  var myElm = document.getElementById('my-elm');
+  var shadow = myElm.attachShadow({ mode: 'open' });
+  shadow.innerHTML = '<b id="my-elm"><slot /></b>';
```
Comment (kasperisager) [inline _drafts/SC4-1-1-unique-id.md]: The self-closing `<slot>` still needs a fix.

File: _drafts/SC4-1-1-unique-id.md (hunk 4 above)
Comment (Brynanders) [reply _drafts/SC4-1-1-unique-id.md]: Added

**Comment by Brynanders** [comment]
@WilcoFiers added `display:non` test case
@kasperisager added closing `</slot>`
@carlosapaduarte removed H74

**Comment by jkodu** [comment]
Final PR - https://github.com/auto-wcag/auto-wcag/pull/134
